design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+ # Observe Only Resources
+
+* Owners: Hasan Turken (@turkenh)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+When using Crossplane to manage resources, we typically create a Managed resource that represents the desired state of
+the resource at a provider. Crossplane takes ownership of the resource, starts acting as the source of truth, and
+ensures configuration matches the desired state.
+
+Sometimes, we want to "observe" an existing resource without taking ownership of it. This could be useful in several
+scenarios, which could be grouped as follows:
+
+- Referencing existing resources without managing them
+  - In your composition, you want to reference network resources like VPC and subnets that are managed by another tool
+  or team.
+- Fetching data from existing resources
+  - You need information about an existing VPC, such as its CIDR range and the subnets it contains.
+- Gradual migration of existing/legacy infrastructure to Crossplane
+  - You have existing infrastructures managed by Terraform, and you want to migrate them gradually to Crossplane.
+  - You have a legacy infrastructure that you want to migrate to Crossplane, but you want to test the managed resources
+  before taking ownership of the underlying resources.
+  - For an existing resource, you don’t want to give some initial configuration that might override the actual
+  configuration. You want to late-initialize all fields, including the ones that would be required otherwise.
+- Only observing some fields after the initial creation
+  - You want to create an EKS Node Group with a scaling configuration where you configured an initial desired size.
+  After the creation, you want to ignore changes in the size which is now being controlled by the cluster autoscaler.
+
+Currently, Crossplane does not have a built-in way of observing resources without taking ownership of them. There are
+two workarounds used by the community as an interim solution for this gap:
+
+- Using the provider-terraform to observe resources with the help of Terraform data sources.
+- Wrapping resources with provider-kubernetes to observe resources managed by Crossplane but as a shared object between
+multiple Compositions.
+
+In this document, we aim to introduce a solution to observe resources with Crossplane without taking ownership of them.
+This would allow users to integrate existing cloud resources with the Crossplane ecosystem without giving full ownership.
+
+## Goals
+
+- Introduce a way to observe existing resources without taking ownership of them in Crossplane.
+- Enable seamless integration of existing cloud resources with the Crossplane ecosystem.
+- Allow users to fetch data from existing resources without giving full ownership.
+
+### Non-goals
+
+- Partially managing a resource by observing a subset of fields.
```

Comment:
Yes. 
I need to add some explanation here on why this is a non-goal, but, here is some context from a related [slack discussion](https://crossplane.slack.com/archives/CEG3T90A1/p1670356012575189?thread_ts=1670285697.486339&cid=CEG3T90A1):

> I’ve put some thought into this; while it sounds like partially observing a resource initially, I believe it has a fundamental difference that is, here, we want to use those parameters during creation. Whereas we expect observe only to be a completely read-only operation that should never make any changes on the external system including the creation.

> I believe it deserves a dedicated issue in Crossplane, but, I can imagine something like the following could do the trick:

> ```yaml
> apiVersion: eks.aws.upbound.io/v1beta1
> kind: NodeGroup
> metadata:
>   name: sample-eks-ng
> spec:
>   ignoreChanges:
>     - scalingConfig[0].desiredSize
>   forProvider:
>     # other fields
>     scalingConfig:
>       - desiredSize: 1
>         maxSize: 1
>         minSize: 1
> ```

Reply:
Added some explanation to the doc on why this is a non-goal.

Reply:
I like the `spec.initProvider` approach and I believe it could be good alternative to the one proposed in https://github.com/crossplane/crossplane/issues/3516. 

However, as I stated [here](https://github.com/crossplane/crossplane/pull/3531#discussion_r1087895081), I believe this should be tackled separately.